### PR TITLE
test: fix EditableMetadata test flake

### PR DESCRIPTION
**Pull request by EmilyBonar**
## Description
Fixes a flaky test. Latest version of test in this PR passed CI 4/4 times tested.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.
- [ ] If modifying `/webui/react/src/shared/` verify `make -C webui/react test-shared` passes.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by hkang1 — APPROVED** [review]
LGTM! I reran it a bunch of times on the CI and I wasn't able to trigger a flake.

File: webui/react/src/components/Metadata/EditableMetadata.test.tsx [hunk 1]
```diff
@@ -55,17 +55,20 @@ describe('EditableMetadata', () => {
...
     const resultMetadata = Object.fromEntries(metadataArray.filter(
-      ([ key, value ]) => key !== removalMetadata[0] && value !== removalMetadata[1],
+      (_metadata, idx) => idx !== removalIndex,
     ));
-    const { handleOnChange } = setup(initMetadata, true);
+    const { handleOnChange, view } = setup(initMetadata, true);
+
+    const actionButton = (view.getAllByRole('button', { name: 'action' }))[removalIndex];
+    await user.click(actionButton);
+    user.click(await view.findByText('Delete Row', undefined, { container: actionButton }));
```
[the remark this answers is not in the document]
Comment (EmilyBonar) [reply webui/react/src/components/Metadata/EditableMetadata.test.tsx:68]: When I try removing it and testing locally it still passes but returns a warning about overlapping `act()` calls, so I'm going to leave it in

[the remark this answers is not in the document]
Comment (hkang1) [reply webui/react/src/components/Metadata/EditableMetadata.test.tsx:68]: Sorry I meant changing it to (adding an `await` in front):
```
await user.click(await view.findByText(...));
```

[the remark this answers is not in the document]
Comment (EmilyBonar) [reply webui/react/src/components/Metadata/EditableMetadata.test.tsx:68]: Ah, I see. No, removing that await is what got the test to pass. I'm not 100% sure on why, but it seems that with the await in front it was trying to refer to the element returned by `await view.findByText('Delete Row', undefined, { container: actionButton })` after it had already been deleted, causing the error.

**Pull request merged**